google/cloud/pubsub/publisher.h

```diff
@@ -189,6 +185,19 @@ class Publisher {
     connection_->ResumePublish({std::move(ordering_key)});
   }
 
+  /// @deprecated Use `Publisher(connection)` instead.
+  GOOGLE_CLOUD_CPP_DEPRECATED("use `Publisher(connection)` instead")
+  explicit Publisher(std::shared_ptr<PublisherConnection> connection,
+                     std::initializer_list<internal::NonConstructible>)
+      : Publisher(std::move(connection)) {}
```

Comment:
optional: At the moment, this is unnecessary, as `Publisher(conn, {})` will hit the `PublisherOptions` overload. It only comes into play once we add `Publisher(std::shared_ptr<PublisherConnection> connection, Options options)`.

We can keep it though.

Reply:
Removed.